lib/open_food_network/address_finder.rb

```diff
@@ -74,9 +89,11 @@ def last_used_ship_address
     # Assumption: front-end users can't ask this library for an address using
     # a customer or user other than themselves...
     def allow_search_by_email?
-      return false unless email.present? && (user.present? || customer.present?)
-      return false unless email == customer.andand.email || email == user.andand.email
-      true
+      email.present? && email_matches_customer_or_user?
+    end
+
+    def email_matches_customer_or_user?
+      email == customer.andand.email || email == user.andand.email
```

Comment:
is it really possible for `customer` and `user` records to have an empty email? I would expect them to have them mandatory even at DB level with a `NOT NULL` constraint.

This code shouldn't be concerned about data consistency. This should be provided elsewhere in the code.

Reply:
It's not testing whether they have an empty email. Its testing whether `customer` or `user` are present at all in the local context.

`customer.andand.email` is shorthand for `customer && customer.email`

Reply:
Oops, I see.